ophyd/ophydobj.py

```diff
@@ -121,6 +124,14 @@ def __init__(self, *, name=None, attr_name='', parent=None, labels=None,
             name = self.name
         # Instantiate logger
         self.log = logging.getLogger(base_log + '.' + name)
+        self.__register_instance(self)
```

Comment:
this should go in `__init_subclass__` so sub-classes can not escape :smiling_imp:

Reply:
Hm. `__init_subclass__` doesn't have access to the instance, does it? Are you envisioning redefining `__init__` in `__init_subclass__`?

Reply:
ah, nvm